ci: add Azure Pipelines

cirulls opened the pull request:
## Summary

This PR introduces [Azure Pipelines](https://azure.microsoft.com/en-gb/services/devops/pipelines/) as an alternative CI for running our test suite under Windows. 

## Why

Appveyor does not allow to run parallel builds with its free version. This means that we have to wait up to 40 minutes to have results from AppVeyor compared to 10 minutes on Travis (which can run parallel builds). 

## What has changed

I set up an [account on Azure Devops](https://dev.azure.com/sandrocirulli/sandrocirulli/_build?definitionId=1). This is linked to my own account required by Microsoft but the builds should be visible to anyone. 

The new CI relies on an `azure-pipelines.yml` sitting at the root of the xspec repository, this works in a similar way as the `.travis.yml` and `appveyor.yml` files and runs the same tests in the same matrix configuration. I introduced the changed suggested by @AirQuick in #629 as they greatly smooth the inconsistencies between AppVeyor and Azure Pipelines running environments. 

I also introduced the Azure Pipelines badge in the README file which shows the status of the build and provides a link to the latest build.

## How I tested it 

- All the tests are green in [build 112](https://dev.azure.com/sandrocirulli/sandrocirulli/_build/results?buildId=112). This is the latest commit in this branch.
- A failed test in the bats-like test is correctly caught in [build 107](https://dev.azure.com/sandrocirulli/sandrocirulli/_build/results?buildId=107).
- A failed test in the java compilation class is correclty caught in [build 108](https://dev.azure.com/sandrocirulli/sandrocirulli/_build/results?buildId=108).
- A failed test in the .xspec tests is correclty caught in [build 109](https://dev.azure.com/sandrocirulli/sandrocirulli/_build/results?buildId=109).
- A failed test in the end-to-end tests is correclty caught in [build 111](https://dev.azure.com/sandrocirulli/sandrocirulli/_build/results?buildId=111).

The three parallel builds take a total time of circa 12 minutes to run which is quicker than the 40 minutes we usually have to wait for sequential builds on AppVeyor. Azure Pipelines is also able to run builds from different commits in parallel as shown in this screenshot (three builds are running independently at the same time):

![image](https://user-images.githubusercontent.com/13256236/65061673-58c02f80-d972-11e9-9b05-9a78ac2de133.png)

AirQuick commented:
Thanks for setting this up!

Not directly related to this pull request. Is it possible to change the organization name from `sandrocirulli` to `xspec`? (`https://dev.azure.com/xspec/`)
Maybe [this](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/rename-organization)?

AirQuick commented:
And the project name (dev.azure.com/sandrocirulli/**sandrocirulli**/). ([this](https://docs.microsoft.com/en-us/azure/devops/organizations/projects/rename-project?view=azure-devops)?)

cirulls commented:
@AirQuick: I renamed the organisation and the project in Azure DevOps and updated the README file,  Azure Pipelines for XSpec is now available [here](https://dev.azure.com/xspec/xspec/_build?definitionId=1&_a=summary).

AirQuick commented on README.md:
```diff
@@ -1,5 +1,6 @@
 [![Travis Build Status](https://api.travis-ci.org/xspec/xspec.svg?branch=master "Travis Build Status")](https://travis-ci.org/xspec/xspec)
 [![AppVeyor Build status](https://ci.appveyor.com/api/projects/status/github/xspec/xspec?branch=master&svg=true "AppVeyor Build Status")](https://ci.appveyor.com/project/xspec/xspec/branch/master)
+[![Azure Pipelines Build Status](https://dev.azure.com/xspec/xspec/_apis/build/status/xspec.xspec?branchName=azure_pipelines_test)](https://dev.azure.com/xspec/xspec/_build/latest?definitionId=1&branchName=azure_pipelines_test)
```
This URL scheme and hier-part look fine, but `branchName=azure_pipelines_test` looks suspicious. `=master`, perhaps?

cirulls replied:
Thanks for spotting this, I used the branch of this pull request for showing the badge and the link but forgot to change it to `master` before initiating this pull request. This is now done in 
990c307. Let me know if you spot anything else.

AirQuick commented:
The build went well and finished earlier than Travis. The status badge turned green.
@cirulls has already [verified](https://github.com/xspec/xspec/pull/630#issue-318432055) that failed tests will be correctly caught, which really convince us. Thanks for making this possible!

cirulls commented:
Thanks for merging it, I hope it's going to be useful and eliminate some of the waiting time caused by AppVeyor.
